### ISSUE TITLE
Feature request: Unified games system, savestates and savefiles directories

### ISSUE THREAD

**Issue by Wintermute0110** [xbmc/xbmc#14882]
### Describe the bug

The `system` directory is where the BIOS and other important files needed by the Game cores are placed. Currently each addon requires to put the BIOSse on a directory USER_DATA/game.libretro.core_name/resources/system. As discussed in [this](https://forum.kodi.tv/showthread.php?tid=332966&pid=2788492#pid2788492) and subsequent posts with @garbear, this is a design flaw because many BIOSes are shared among cores. With the current implementation this is not possible. The system directory is a property of the libretro frontend, not of the cores.

Another problem is that the savestates are created next to each ROM. ROM directories must not be writable by the libretro frontend (that is, Kodi).

## Expected Behavior

A better implementation is to create the system, savestates and savefiles directories in a special directory in USER_DATA. For example:

```
profile://user_data/games/system/ --> system directory
profile://user_data/games/savefiles/core_name/ROM_name
profile://user_data/games/savestates/core_name/ROM_name
profile://user_data/games/info/ --> core info files.
```

Kodi core is expected to create these directories and use them instead of the current directories.

## Current Behavior

Explained in the bug description. The current implementation works, however it complicates a lot the setup of emulators in Kodi Leia. The proposed implementation is based on Retroarch behaviour, it's more efficient in terms of disk space and easy to set up.

## Possible Fix

Explained in the Expected behaviour.

### Your Environment
Used Operating system: tested on Debian Linux amd64 architecture with a custom-built Kodi version cloning Github repository.

**Comment by MartijnKaijser** [comment]
Feature request belong on the forum

**Comment by enen92** [comment]
Or in this case maybe you want to bring it as an issue to https://github.com/garbear/xbmc

**Comment by MartijnKaijser** [comment]
Garbear was planning to move his issues here so that's pointless imo (unless he changed his mind)

**Comment by Wintermute0110** [comment]
@MartijnKaijser, @garbear told me to write it here ...

@enen92 garbear branch is merged into master right?

**Comment by Wintermute0110** [comment]
@MartijnKaijser so, Retroplayer will be developed exclusively by garbear? No Retroplayer issues in Kodi master branch????

**Comment by MartijnKaijser** [comment]
Is your keyboard stuck on ??????

**Comment by Wintermute0110** [comment]
@MartijnKaijser No, but it will help if you speak the things clearly and **answer** to the questions properly. Do you have a code of conduct? It took me 20 minutes to write this report (plus the time to think about the solution), it took 10 seconds to close it without any reasonable explanations beyond "I close it because I want to close it".

**Comment by garbear** [comment]
I've identified that the current behavior diverges from the libretro API. libretro specifies a single system directory: https://github.com/kodi-game/game.libretro/blob/af15fa7/src/libretro/libretro.h#L532

I don't know if non-conformance is considered a "bug", but it is certainly an issue that needs to be fixed, so if possible I'd like to track the problem here instead of making @Wintermute0110 shuffle this information around even more.

**Comment by garbear** [comment]
Forum thread for non-development related discussion: https://forum.kodi.tv/showthread.php?tid=337526

**Comment by garbear** [comment]
Progress is being made on this here: https://github.com/garbear/xbmc/issues/158
